java/rocksjni/sst_file_writerjni.cc

```diff
@@ -110,6 +110,43 @@ void Java_org_rocksdb_SstFileWriter_merge(JNIEnv *env, jobject jobj,
 
 /*
  * Class:     org_rocksdb_SstFileWriter
+ * Method:    merge
+ * Signature: (J[B[B)V
+ */
+void Java_org_rocksdb_SstFileWriter_merge__J_3B_3B(JNIEnv *env, jobject jobj,
+                                                   jlong jhandle, jbyteArray jkey,
+                                                   jbyteArray jval) {
+
+  jbyte* key = env->GetByteArrayElements(jkey, nullptr);
+  if(key == nullptr) {
+    // exception thrown: OutOfMemoryError
+    return;
+  }
+  rocksdb::Slice keySlice(
+      reinterpret_cast<char*>(key),  env->GetArrayLength(jkey));
```

Comment:
@sagar0 Two reasons:

1. Passing in the `length` is likely cheaper than calling `env->GetArrayLength`
2. Passing in the `length` and `offset` allow you to just store part of an existing byte array. Users have previously requested being able to store part of an array without them needing to copy to a smaller array first. This makes sense when they have many keys/values encoded into a single byte array.